### ISSUE TITLE
Insight into promisifyAll bug?

### ISSUE THREAD

**Issue by tnrich** [tgriesser/knex#1010]
Hey there, when running our app (connected to an oracle db backend currently), we occasionally get the following error:

```
/Users/tnrich/Sites/ve/api_server_bookshelf/node_modules/bluebird/js/main/promisify.js:280
        throw new TypeError("the target of promisifyAll must be an object or a
              ^
TypeError: the target of promisifyAll must be an object or a function

    See http://goo.gl/9ITlV0

    at Function.Promise.promisifyAll (/Users/tnrich/Sites/ve/api_server_bookshelf/node_modules/bluebird/js/main/promisify.js:280:15)
    at /Users/tnrich/Sites/ve/api_server_bookshelf/node_modules/knex/lib/dialects/oracle/index.js:77:17
```

Does anyone have any ideas/recommendations for fixing it? Thanks!

**Comment by vschoettke** [comment]
There is probably a problem connecting to the db. The promisifyAll was called before the check if the connection had an error. This was already fixed in #905. But no updated version has been published yet.

**Comment by tnrich** [comment]
Awesome thanks for the info @vschoettke! Any idea when a new version might be published?
